### PR TITLE
add a shell option to NPM class

### LINKED ISSUE #12: Unclear if NPM or file cannot be found: [WinError 2] The system cannot find the file specified
Package version (if known): 0.1.2

## Describe the bug
When I try to run pkg.install(), I get the error "[WinError 2] The system cannot find the file specified"

I suspect it can't find npm vs not being able to find the file.  I'd expect it to tell me it can't find npm to make it clear.

But NPM is findable in my path.  I even tried manually appending the NPM bin folder to the path without success.  And I can run NPM from the command line.  So I'm at a loss as to what the issue is.

## Steps to Reproduce

# install npm and add to path
pip install pynpm

```
import os
import sys
from pynpm import NPMPackage

def install_dependecies():
    npm_path = r'C:\Program Files\nodejs\node_modules\npm\bin'
    sys.path.append(npm_path)
    executable = 'npm.cmd'
    for p in os.environ["PATH"].split(os.pathsep):
        if os.path.exists(os.path.join(p, executable)):
            print('Found NPM')

    #jquery_pkg_path = os.getcwd() + r'\3rdparty\jquery-ui-1.12.1.custom\package.json'
    #jquery_pkg_path = os.getcwd() + '/3rdparty/jquery-ui-1.12.1.custom/package.json'
    #jquery_pkg_path = './3rdparty/jquery-ui-1.12.1.custom/package.json'
    #jquery_pkg_path = r'C:\Users\user\source\repos\myproject\3rdparty\jquery-ui-1.12.1.custom\package.json'
    jquery_pkg_path = 'C:/Users/user/source/repos/myproject/3rdparty/jquery-ui-1.12.1.custom/package.json'
    if os.path.exists(jquery_pkg_path) == True:
        print('File exists: ' + jquery_pkg_path)
    else:
        print('File does not exist: ' + jquery_pkg_path)
    jquery_pkg = NPMPackage(jquery_pkg_path)
    jquery_pkg.install()
```

## Expected behavior
I expect it to tell me whether it's failing to find NPM or my file.  I suspect it is failing to find NPM.

## Screenshots (if applicable)
I get the following output:
> Found NPM
> File exists: C:/Users/user/source/repos/myproject/3rdparty/jquery-ui-1.12.1.custom/package.json
> C:/Users/user/source/repos/myproject/3rdparty/jquery-ui-1.12.1.custom/package.json
> 2021-02-22 09:34:49,469 ERROR    [Main.py:59 main] [WinError 2] The system cannot find the file specified
> Traceback (most recent call last):
>   File "C:\Users\user\source\repos\myproject\Main.py", line 57, in main
>     api.treeview_dictionary_editable.install_dependecies()
>   File "C:\Users\user\source\repos\myproject\api\treeview_dictionary_editable.py", line 17, in install_dependecies
>     jquery_pkg.install()
>   File "C:\Users\user\source\repos\myproject\env\lib\site-packages\pynpm\package.py", line 76, in _run_npm
>     **kwargs
>   File "C:\Users\user\source\repos\myproject\env\lib\site-packages\pynpm\utils.py", line 25, in run_npm
>     cwd=pkgdir,
>   File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 339, in call
>     with Popen(*popenargs, **kwargs) as p:
>   File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 800, in __init__
>     restore_signals, start_new_session)
>   File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\subprocess.py", line 1207, in _execute_child
>     startupinfo)
> FileNotFoundError: [WinError 2] The system cannot find the file specified
> [WinError 2] The system cannot find the file specified
> Stack trace:
>  >  File "C:\Users\user\source\repos\myproject\Main.py", line 77, in <module>
>  >    main()

### LINKED ISSUE #14: The lib is not compatible with Windows
Would it be possible to make the lib compatible with windows ? 
I checked #12 and it seems this is a general issue for windows users but as the lib has no CI on this platform I cannot say if it's me or broader issue...

### PR DESCRIPTION
**Pull request by 12rambau**
Fix #12 

As mentioned in the related issue, the error is comming from the windows systems not being able to find the NPM command. 
I propagated the changes suggested by @travisjwalter and updated the doc accordingly.

Fix #14 

To check that my changes are making sense I also added a windows test in the test suit.

### CODE REVIEW

**Comment by lnielsen** [comment]
Sorry, we've been extremely busy launching a new update of Zenodo, so didn't have time to check. We've added it on the review list, so we'll check it out soon. Sorry for the long wait

**Comment by lnielsen** [comment]
The CI got disabled due to inactivity, so I've just enabled it. Can I ask you to repush so CI will run again.

**Comment by slint** [comment]
@12rambau there seems to be an error with the updated CI file: https://github.com/inveniosoftware/pynpm/actions/runs/6922649852/workflow

Can you have a look? I think the `matrix.os` variable needs to be a list.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -26,13 +26,16 @@ on:
 
 jobs:
   Tests:
-    runs-on: ubuntu-20.04
     timeout-minutes: 20
     strategy:
       matrix:
+          os: ubuntu-20.04
```
Comment (12rambau) [inline .github/workflows/tests.yml]:
```suggestion
          os: [ubuntu-20.04]
```

**Comment by 12rambau** [comment]
@slint, Done, I just need your approval again to launch the workflow.

**Comment by slint** [comment]
@12rambau I'm sorry, GitHub actions seem to be squeezing out the errors one-by-one after each push. Now it seems to be failing because of the `matrix.include.os` that needs to be a list as well. Can you push again with this change?

**Comment by 12rambau** [comment]
I modified it from the GIthub interface to get some feedback from their editor. `os` can (and should) be a string it's the content of `include` that needs to be a list, a list of `{"os", "python"}` dict. 

Normally this time everything should be ok (syntactically speaking)

**Comment by slint** [comment]
@12rambau looks like we have lift-off on the GH Actions; it seems like the PR is only failing on the black formatting (which is good news :) ). 

I've also changed a setting on the repo regarding approval to run workflows, which might improve the feedback cycle.

**Review by slint — APPROVED** [review]
Thanks a lot for all the work! LGTM, so I'll squash and merge

**Comment by slint** [comment]
@12rambau, one last question before I release `v0.2.0` for `master`: what should the Copyright be for your contribution? E.g. if this is a personal contribution, that would just be `Copyright (C) 2023 Rambaud Pierrick`, or if done as part of your work for your organization then we'd need something like `Copyright (C) 2023 Food and Agriculture Organization` or similar.

**Comment by 12rambau** [comment]
personal contribution 100%

**Pull request merged**